test/e2e_node/memory_manager_test.go

```diff
@@ -802,7 +936,757 @@ var _ = SIGDescribe("Memory Manager Incompatibility Pod Level Resources", framew
 				initCtnParams = []memoryManagerCtnAttributes{}
 			})
 
-			ginkgo.It("should not report any memory data during request to pod resources List when pod has pod level resources", func(ctx context.Context) {
+			ginkgo.It("scope: pod, should allocate exclusive memory to a guaranteed pod with pod-level resources and guaranteed container, PodLevelResourceManagers enabled", func(ctx context.Context) {
+				currentCfg, err := getCurrentKubeletConfig(ctx)
+				framework.ExpectNoError(err)
+				updateKubeletConfigIfNeeded(ctx, f, configureMemoryManagerInKubelet(currentCfg, &memoryManagerKubeletArguments{
+					policyName:                     string(staticPolicy),
+					topologyManagerPolicy:          topologymanager.PolicyBestEffort,
+					topologyManagerScope:           topologymanager.PodTopologyScope,
+					enablePodLevelResources:        true,
+					enablePodLevelResourceManagers: true,
+				}))
+
+				ctnParams = []memoryManagerCtnAttributes{
+					{
+						ctnName: "gu-container",
+						cpus:    "100m",
+						memory:  "128Mi",
+					},
+				}
+				testPod = makeMemoryManagerPod("gu-pod-level-gu-ctn", nil, ctnParams)
+				testPod.Spec.Resources = &v1.ResourceRequirements{
+					Limits: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("100m"),
+						v1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+					Requests: v1.ResourceList{
+						v1.ResourceCPU:    resource.MustParse("100m"),
+						v1.ResourceMemory: resource.MustParse("128Mi"),
+					},
+				}
+
+				ginkgo.By("Running the test pod")
+				testPod = e2epod.NewPodClient(f).CreateSync(ctx, testPod)
+
+				if !*isMultiNUMASupported {
+					return
+				}
+
+				verifyMemoryPinning(f, ctx, testPod, []int{0})
```

Comment:
I realized only now what this check actually does, and it's fragile. We had the same problem on cpu manager tests. We should not check all allocations are aligned to the exact NUMA node X (this is the fragile part as we guess what the allocation would do). We should check all the allocations are aligned on the same node, whatever it is, or that all the allocations are aligned to a set of nodes whose cardinality is Z (or up to Z). Can and probably should be fixed in another PR.